tmt/steps/report/reportportal.py

```diff
@@ -39,112 +36,169 @@ class ReportReportPortalData(tmt.steps.report.ReportStepData):
-        bytestream = io.BytesIO()
-        with zipfile.ZipFile(bytestream, "w", compression=zipfile.ZIP_DEFLATED,
-                             compresslevel=1) as zipstream:
-            # XML file names are irrelevant to ReportPortal
-            with zipstream.open("tests.xml", "w") as entry:
-                entry.write(data.encode("utf-8"))
-        bytestream.seek(0)
-
-        # Send the report to the ReportPortal instance
+        assert self.step.plan.name is not None
+        launch_name = self.get("launch") or self.step.plan.name
+
+        version = "v1"
+        url = f"{server}/api/{version}/{project}"
+        headers = {"Authorization": "bearer " + token,
+                   "accept": "*/*",
+                   "Content-Type": "application/json"}
+
+        TMT_env_vars = self.get("env-vars") or []
+        attributes = [{'key': k, 'value': v[0]} for k, v in self.step.plan._fmf_context.items()]
+
+        # Communication with RP instance
         with tmt.utils.retry_session() as session:
-            url = f"{server}/api/v1/{project}/launch/import"
-            self.debug(f"Send the report to '{url}'.")
-            response = session.post(
-                url,
-                headers={
-                    "Authorization": "bearer " + token,
-                    "accept": "*/*",
-                    },
-                files={
-                    # The zip filename is used as the launch name in ReportPortal
-                    "file": (launch_name + ".zip", bytestream, "application/zip"),
-                    },
-                )
-
-        # Handle the response
-        try:
-            message = tmt.utils.yaml_to_dict(response.text).get("message")
-        except (tmt.utils.GeneralError, KeyError):
-            message = response.text
-        if not response.ok:
-            raise tmt.utils.ReportError(
-                f"Received non-ok status code from ReportPortal, response text is: {message}")
-        else:
-            self.debug(f"Response code from the server: {response.status_code}")
-            self.debug(f"Message from the server: {message}")
-            self.info("report", "Successfully uploaded.", "yellow")
+
+            # Create a launch
+            self.info("launch", launch_name, color="green")
+            response = session.post(url=f"{url}/launch",
+                                    headers=headers,
+                                    json={"name": launch_name,
+                                          "description": self.step.plan.summary,
+                                          "attributes": attributes,
+                                          "startTime": timestamp()})
+            self.handle_response(response)
+            launch_uuid = tmt.utils.yaml_to_dict(response.text).get("id")
+            assert launch_uuid is not None
+            self.info("    uuid", launch_uuid, "yellow")
+
+            # For each test
+            for result in self.step.plan.execute.results():
```

Comment:
I see you are uploading the results after they already finished. 
Can we upload the results of each test separately immediately when it is finished, in this case, we will get online results,
E.g.:
1. Open RP branch on TMT start
loop over TMT tests:
  2. Open the RP test before executing the test
  3. Close the RP test after the test ended
4. Close  RP launch

Note:
In any case, we must keep and provide the JUnit results, which include a full report

Reply:
Not at this moment, the `execute` and `report` steps are isolated, and there are no before testing/before test/after test hooks for `report` plugins to use. It is an interesting idea though, could you file an issue for it?

Adding support for such events & hooks should be made in a general fashion, for other plugins to use when needed, and therefore in a stand-alone PR (or several PRs, this might get too complicated to fit into a single patch).

Reply:
@happz As you suggested I open this discussion: https://github.com/teemtee/tmt/discussions/2092